app/services/leaderboardService.ts

```diff
@@ -1,37 +1,293 @@
+  queryName: string,
+): Promise<string[]> {
+  const queryString = Object.keys(searchQuery)
+    .map(
+      (key) => `${key}=${encodeURIComponent(JSON.stringify(searchQuery[key]))}`,
+    )
+    .join("&");
+
+  const response = await fetch(
+    `${baseUrl}/search/advanced/profiles?${queryString}`,
+    {
+      headers: {
+        Accept: "application/json",
+        "X-API-KEY": apiKey,
+      },
+    },
+  );
+
+  if (!response.ok) {
+    const errorText = await response.text();
+    throw new Error(`${queryName} search failed: ${errorText}`);
+  }
+
+  const data = await response.json();
+  const profileIds =
+    data.profiles?.map((profile: { id: string }) => profile.id) || [];
+
+  return profileIds;
+}
+
+/**
+ * Internal helper to get boosted profiles from external API using OR logic
+ * Returns profiles that have EITHER talent_protocol_talent_holder ≥ 100 OR talent_vault ≥ 100
+ */
+async function getBoostedProfilesViaSearch(apiKey: string): Promise<string[]> {
+  const baseUrl = "https://api.talentprotocol.com";
+  const boostedProfileIds = new Set<string>();
+
+  // Query 1: talent_protocol_talent_holder ≥ 100 (optimized for top 200)
+  const query1 = {
+    query: {
+      score: {
+        min: 120, // Optimized: focus on profiles likely to be in top 200
+        scorer: "Creator Score",
+      },
+      credentials: [
+        {
+          slug: "talent_protocol_talent_holder",
+          valueRange: { min: BOOST_CONFIG.TOKEN_THRESHOLD },
+        },
+      ],
+    },
+    sort: {
+      score: { order: "desc", scorer: "Creator Score" },
+      id: { order: "desc" },
+    },
+    per_page: 200,
   };
+
+  // Query 2: talent_vault ≥ 100 (optimized for top 200)
+  const query2 = {
+    query: {
+      score: {
+        min: 120, // Optimized: focus on profiles likely to be in top 200
+        scorer: "Creator Score",
+      },
+      credentials: [
+        {
+          slug: "talent_vault",
+          valueRange: { min: BOOST_CONFIG.TOKEN_THRESHOLD },
+        },
+      ],
+    },
+    sort: {
+      score: { order: "desc", scorer: "Creator Score" },
+      id: { order: "desc" },
+    },
+    per_page: 200,
+  };
+
+  // Execute both queries in parallel and handle partial failures gracefully
+  const [results1, results2] = await Promise.allSettled([
+    executeSearchQuery(
+      baseUrl,
+      query1,
+      apiKey,
+      "talent_protocol_talent_holder",
+    ),
+    executeSearchQuery(baseUrl, query2, apiKey, "talent_vault"),
+  ]);
+
+  // Process results and deduplicate
+  if (results1.status === "fulfilled") {
+    results1.value.forEach((id) => boostedProfileIds.add(id));
+  }
+
+  if (results2.status === "fulfilled") {
+    results2.value.forEach((id) => boostedProfileIds.add(id));
+  }
+
+  const finalResults = Array.from(boostedProfileIds);
+  return finalResults;
+}
+
+/**
+ * Service function to get boosted profiles - called by API route
+ */
+export async function getBoostedProfilesData(): Promise<string[]> {
```

Comment:
I don't think you need this wrapper, since it just calls `getBoostedProfilesViaSearch` but adding an API Key